types/office-js-preview/index.d.ts

```diff
@@ -573,6 +573,59 @@ declare namespace Office {
         onVisibilityModeChanged(
             listener: (message: VisibilityModeChangedMessage) => void,
         ): Promise<() => Promise<void>>;
+
+        /**
+         * Represents a modal notification dialog that can appear when the user attempts to close a document. The document won't close until the user responds.
+         * 
+         * @remarks
+         * [Api set: SharedRuntime BETA (PREVIEW ONLY)]
+         * @beta
+        */
+        beforeDocumentCloseNotification: BeforeDocumentCloseNotification;
+    }
+    /**
+     * Represents a modal notification dialog that can appear when the user attempts to close a document. The document won't close until the user responds.
+     * The notification dialog will allow the user to confirm the request to close the document or cancel the request to close the document.
+     * 
+     * @remarks
+     * [Api set: SharedRuntime BETA (PREVIEW ONLY)]
+     * @beta
+    */
+    interface BeforeDocumentCloseNotification {
+        /**
+         * Enable a modal notification dialog that appears when the user attempts to close a document. The document won't close until the user responds.
+         * This notification dialog asks the user to confirm the request to close the document, or allows the user to cancel the request to close the document.
+         * 
+         * @remarks
+         * [Api set: SharedRuntime BETA (PREVIEW ONLY)]
+         * @beta
+         */
+        enable(): Promise<void>;
+
+        /**
+         * Prevents the notification dialog from appearing when the user attempts to close a document.
+         * 
+         * @remarks
+         * [Api set: SharedRuntime BETA (PREVIEW ONLY)]
+         * @beta
+         */
+        disable(): Promise<void>;
+
+        /**
+         * Adds a listener that detects when the `BeforeDocumentCloseNotification` close operation is cancelled. 
```

Comment:
Please use "handler" or "event handler" in place of "listener" throughout this API, including the signature. 

(For some reason this comment didn't get saved when I reviewed last night.)

Reply:
This API follows the same pattern as onVisibilityModeChanged (directly preceding, see line 573), including "listener" in the signature. Do you still think that I should change listener to handler for onCloseActionCancelled?

Reply:
> This API follows the same pattern as onVisibilityModeChanged (directly preceding, see line 573), including "listener" in the signature. Do you still think that I should change listener to handler for onCloseActionCancelled?

@alison-mk I see. That's not the correct term, and the conceptual topic for onVisbilityModeChanged ([Handle the visibility changed event](https://docs.microsoft.com/en-us/office/dev/add-ins/develop/show-hide-add-in#handle-the-visibility-changed-event)) correctly calls it a "handler". Let's not introduce an inconsistency. So, you can either change both the events to "handler" in this PR or leave them both as "listener" now and create a work item for the team to fix this in the future. Let me know if you prefer the latter alternative and I'll approve. Thanks.

Reply:
I feel uncertain about changing a type definition signature without consulting an engineer. I'm not sure what the ripple effects might be. I can change "listener" to "event handler" in the description though.

Reply:
> I feel uncertain about changing a type definition signature without consulting an engineer. I'm not sure what the ripple effects might be. I can change "listener" to "event handler" in the description though.

OK. I'll approve that, and I'll make a work item to fix this.